addons/l10n_tw_edi_ecpay/models/account_move.py

```diff
@@ -0,0 +1,297 @@
+            "Items": self._l10n_tw_edi_prepare_item_list()
+        }
+        if self.l10n_tw_edi_is_donate:
+            json_data.update({"LoveCode": self.l10n_tw_edi_love_code})
+        return json_data
+
+    def _transfer_time(self, time_before):
+        time_after = (
+            datetime.datetime.strptime(time_before, "%Y-%m-%d %H:%M:%S") - datetime.timedelta(hours=8)
+        ).strftime("%Y-%m-%d %H:%M:%S")
+        return time_after
+
+    def _l10n_tw_edi_send(self, json_content):
+        request_url, merchant_id, hashkey, hashIV = self.env.company._l10n_tw_edi_prepare_api_param()
+        # Ensure to lock the records that will be sent, to avoid risking sending them twice.
+        self.env["res.company"]._with_locked_records(self)
+        # prepare EcPayAPI class
+        response_data = EcPayAPI(hashkey, hashIV, merchant_id).call_ecpay_api(request_url + "/Issue", json_content)
+        if response_data.get("RtnCode") != 1:
+            return [_(f"Invoice {self.name} Error: {response_data.get('RtnMsg')}")]
+        self.l10n_tw_edi_ecpay_invoice_id = response_data.get("InvoiceNo")
+        self.l10n_tw_edi_invoice_create_date = self._transfer_time(response_data.get("InvoiceDate").replace("+", " "))
+        self.l10n_tw_edi_state = "invoiced"
+        self._message_log(
+            body=_("The invoice has been successfully sent to Ecpay with Ecpay invoice number '%s'.", response_data.get('InvoiceNo')),
+        )
+
+    def _l10n_tw_edi_fetch_ecpay_invoice(self):
+        request_url, merchant_id, hashkey, hashIV = self.env.company._l10n_tw_edi_prepare_api_param()
+        # Ensure to lock the records that will be sent, to avoid risking sending them twice.
+        self.env["res.company"]._with_locked_records(self)
+        # prepare json data
+        json_data = {
+            "MerchantID": merchant_id,
+            "RelateNumber": self.l10n_tw_edi_related_number,
+        }
+
+        # prepare EcPayAPI class
+        response_data = EcPayAPI(hashkey, hashIV, merchant_id).call_ecpay_api(request_url + "/GetIssue", json_data)
+        if response_data.get('RtnCode') != 1:
+            return [_(f"Invoice {self.name} Error: {response_data.get('RtnMsg')}")]
+        self.l10n_tw_edi_invoice_amount = response_data.get("IIS_Sales_Amount")
+        self.l10n_tw_edi_invoice_valid_status = response_data.get("IIS_Invalid_Status")
+        self.l10n_tw_edi_invoice_issue_status = response_data.get("IIS_Issue_Status")
+        self.l10n_tw_edi_invoice_shop_custom_number = response_data.get("IIS_Relate_Number")
+        self.l10n_tw_edi_remain_refundable_amount = response_data.get("IIS_Remain_Allowance_Amt")
+        self.l10n_tw_edi_state = "valid"
+
+    def _l10n_tw_edi_run_invoice_invalid(self):
+        if self.l10n_tw_edi_ecpay_invoice_id:
+            if self.l10n_tw_edi_invoice_valid_status == "1":
+                raise UserError(_(f"The following invoice: {self.l10n_tw_edi_ecpay_invoice_id} is abolished"))
```

Comment:
You cannot translate fstrings /!\
```suggestion
                raise UserError(_(f"The following invoice: %(invoice_id) is abolished", invoice_id=self.l10n_tw_edi_ecpay_invoice_id))
```